### PR TITLE
ci: add stage for updating stable.json [skip ci]

### PR DESCRIPTION
**Pull request by radnov**
Adds a new stage for updating the stable.json file with the [update-stable-json.py script](https://github.com/dhis2/dhis2-releases/tree/master/tools/update-stable-json).

Example PR created in the `dhis2-releases` repo with a change triggered ([build](https://ci.dhis2.org/job/test-stable/view/tags/job/2.51.2.3/2/)) from a test `dhis2-core` tag `2.51.2.3` - https://github.com/dhis2/dhis2-releases/pull/172

### CODE REVIEW

File: jenkinsfiles/stable [hunk 1]
```diff
@@ -172,6 +172,49 @@ pipeline {
...
+                            --json-file ./downloads/v1/versions/stable.json
+                        """
+
+                        sh 'git config  user.email "$GITHUB_EMAIL"'
+                        sh 'git config  user.name "$GITHUB_USER"'
+
+                        releasesBranch = "add-release-${newImageTagFormat}"
+
+                        sh "git checkout -b $releasesBranch"
+                        sh 'git add .'
```
Comment (teleivo) [inline jenkinsfiles/stable:208]: Maybe just add `stable.json` to be on the safe side 😅

File: jenkinsfiles/stable [hunk 2]
```diff
@@ -172,6 +172,49 @@ pipeline {
...
+                        sh "git diff-index --quiet HEAD || git commit -m \"chore: add version $newImageTagFormat to stable.json\""
+                        sh "git push https://${GITHUB_TOKEN}@github.com/$DHIS2_RELEASES_REPO"
+
+                        sh "gh pr create --head $releasesBranch --fill-first --reviewer Philip-Larsen-Donnelly"
```
Comment (teleivo) [inline jenkinsfiles/stable:212]: Add team devops as reviewers as well?

File: jenkinsfiles/stable [hunk 3]
```diff
@@ -172,6 +172,49 @@ pipeline {
...
         }
 
+        stage('Update stable.json') {
+            when {
+                buildingTag()
+            }
+
+            environment {
+                DHIS2_RELEASES_REPO = "dhis2/dhis2-releases"
+                GITHUB_EMAIL = 'apps@dhis2.org'
```
[the remark this answers is not in the document]
Comment (radnov) [reply jenkinsfiles/stable:182]: Ah, good point! I think Austin has access to that email/user.

@amcgee Can we use a github noreply email for the bot user? Something like this:
image

File: jenkinsfiles/stable [hunk 4]
```diff
@@ -145,7 +145,7 @@ pipeline {
                         imageTag = "$imageTag.0"
                     }
 
-                    def newImageTagFormat = imageTag.replaceFirst(/^2\./, "")
+                    newImageTagFormat = imageTag.replaceFirst(/^2\./, "")
```
[the remark this answers is not in the document]
Comment (radnov) [reply jenkinsfiles/stable:148]: I renamed them to `oldImageTag` and just `imageTag`, as `version` would be a bit ambiguous within the context of the whole script and a lot of other variables would have to be renamed.

**Review by teleivo — APPROVED** [review]

**Review by tonsV2 — APPROVED** [review]

**Pull request merged**